### PR TITLE
Rename to electron

### PR DESCRIPTION
**Pull request by kevinsawicki**
When Atom Shell started over two years ago, its one and only consumer was the [Atom](https://github.com/atom/atom) editor.

And so the original Atom Shell name was just a stopgap to refer to the Chromium/Node integration pieces and the native bits for window management and OS integrations such as the clipboard, the dock, etc.

But over the last two years Atom Shell has grown and matured to become a framework that isn't just for building text editors. People are now using it to build email clients, design tools, chat clients, and video players.

It deserves a better name that decouples it from the Atom editor association and allows it to stand on its own and continue to become the easiest way to build native apps using web tech.

So the new name will be :zap: **electron** :zap: and this pull request will update all the places in the code and the docs to this new name.

/cc @atom/core @atom/non-github-maintainers

### CODE REVIEW

**Comment by thomasjo** [comment]
Awesome! :sparkles:

**Comment by marekhrabe** [comment]
nice one, guys! we love it at @avocode :)

File: atom/browser/default_app/index.html [hunk 1]
```diff
@@ -72,7 +72,7 @@
     };
   </script>
 
-  <h2 style="-webkit-app-region: drag">Welcome to Atom Shell</h2>
+  <h2 style="-webkit-app-region: drag">Welcome to Electron</h2>
 
   <p>
   To run your app with atom-shell, execute the following command under your
```
Comment (50Wliu) [inline atom/browser/default_app/index.html]: Forgot to change it here

File: atom/browser/default_app/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "atom-shell-default-app",
```
Comment (50Wliu) [inline atom/browser/default_app/package.json]: And here?

File: spec/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "atom-shell-specs",
```
Comment (50Wliu) [inline spec/package.json]: And here

**Comment by YuriSolovyov** [comment]
`.asar` makes less sense with Electron, but it is not a huge deal.

**Comment by paulcbetts** [comment]
Yesssssss

**Comment by Tribex** [comment]
Oooh! Sounds nice!
Quark sounds better IMO though.

**Comment by mnquintana** [comment]
@YuriSolovyov Although then we'd end up with `.ear` :stuck_out_tongue_closed_eyes:

**Comment by YuriSolovyov** [comment]
@mnquintana it does not have to be ARchive, it can be a PACKage, or something.

**Comment by mnquintana** [comment]
Now that I think about it, `.elar` wouldn't be bad.

**Comment by YuriSolovyov** [comment]
@kevinsawicki are you going to design a logo too?

**Comment by paulcbetts** [comment]
> Now that I think about it, .elar wouldn't be bad.

I think it'd be _super_ cool if we had as few non-breaking changes to existing code as possible just to change a marketing name.

**Comment by kevinsawicki** [comment]
> are you going to design a logo too?

Eventually yes, a new site is being worked on as well.

**Comment by thedaniel** [comment]
> I think it'd be super cool if we had as few non-breaking changes to existing code as possible just to change a marketing name, so far we're doing pretty good

:point_up: 

It doesn't matter what the archive format is called, .asar is fine forever.

**Comment by thomasjo** [comment]
> It doesn't matter what the archive format is called, .asar is fine forever.

This.

---
If people are super bothered, we can imagine `.asar` being an abbreviation of *Awesome Software ARchive* or whatever.

**Comment by mnquintana** [comment]
> I think it'd be super cool if we had as few non-breaking changes to existing code as possible just to change a marketing name, so far we're doing pretty good

Haha no worries, I was just throwing around names not-so-seriously

**Comment by lee-dohm** [comment]
Posted a reference on Discuss:

https://discuss.atom.io/t/in-case-you-hadnt-heard-atom-shell-is-being-renamed-electron/16098

**Comment by kevinsawicki** [comment]
@zcbenz anything left to do here before shipping this?

Want to add a task list to the pull request body if there are still remaining items?

**Comment by jweinstein** [comment]
I am impressed by this diff.

File: atom/app/atom_main_delegate_mac.mm [hunk 4]
```diff
@@ -19,19 +21,30 @@
...
 }
 
 void AtomMainDelegate::OverrideChildProcessPath() {
-  base::FilePath helper_path =
-      GetFrameworksPath().Append(PRODUCT_NAME " Helper.app")
-                         .Append("Contents")
-                         .Append("MacOS")
-                         .Append(PRODUCT_NAME " Helper");
+  base::FilePath frameworks_path = GetFrameworksPath();
+  base::FilePath helper_path = GetHeleprAppPath(frameworks_path,
```
Comment (paulcbetts) [inline atom/app/atom_main_delegate_mac.mm]: Typo?

**Comment by jprichardson** [comment]
Been using atom-shell for a few months. Electron is an excellent name and is better for marketing. I realize that many devs don't care about this, but it affects us all whether we know it or not. The bigger the community for Electron, the better it is all for all of us. Bravo!

**Comment by kevinsawicki** [comment]
:green_heart: :purple_heart: :blue_heart:

**Comment by SHADOWELITE7** [comment]
nice

**Pull request merged**